### PR TITLE
Convert warning message to debug-level info in spawn plugins

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Suggested by @carmocca and @tchaton 
The current warning is not appropriate, it is really more an info.

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

Part of #1 (it's a lie, this is just here to avoid noisy GitHub bot)

cc @borda

### CODE REVIEW

File: pytorch_lightning/plugins/training_type/tpu_spawn.py [hunk 1]
```diff
@@ -209,15 +209,14 @@ def barrier(self, name: Optional[str] = None) -> None:
             rendezvous(name)
 
     def __transfer_distrib_spawn_state_on_fit_end(self, trainer: "pl.Trainer", results: Any) -> None:
+        rank_zero_debug("Transferring results back to main process ...")
```
Comment (kaushikb11) [inline pytorch_lightning/plugins/training_type/tpu_spawn.py:212]: Tbh, I liked the `"cleaning up tpu spawn environment..."` warning/info message for TPU spawn haha.
So it usually takes a lot of time transferring the results to the main process, and it's a good indicator that this step has been completed.

No strong opinion tho

**Review by kaushikb11 — APPROVED** [review]

File: pytorch_lightning/plugins/training_type/tpu_spawn.py (hunk 1 above)
Comment (awaelchli) [reply pytorch_lightning/plugins/training_type/tpu_spawn.py:212]: which part of the implementation of this function is cleaning things up? As I see it, it is quite the opposite of cleaning up, as it creates a checkpoint file. 

Debug was requested by @carmocca and @tchaton 
I align with you but also I don't care so I will let you all decide

**Review by rohitgr7 — APPROVED** [review]

**Review by Borda — APPROVED** [review]

**Review by justusschock — APPROVED** [review]

File: pytorch_lightning/plugins/training_type/tpu_spawn.py (hunk 1 above)
Comment (carmocca) [reply pytorch_lightning/plugins/training_type/tpu_spawn.py:212]: > As it usually takes a lot of time transferring the results to the main process for TPUs

I'd cater to the most noob users. They don't know that this is slow, or what are "results" referring to, so I'd avoid the message.

But also no strong opinion :P

**Review by carmocca — APPROVED** [review]

**Comment by awaelchli** [comment]
@justusschock @tchaton @rohitgr7 @Borda can you give me your opinion on the [discussion](https://github.com/PyTorchLightning/pytorch-lightning/pull/10864#discussion_r760003947) above.

File: pytorch_lightning/plugins/training_type/tpu_spawn.py (hunk 1 above)
Comment (rohitgr7) [reply pytorch_lightning/plugins/training_type/tpu_spawn.py:212]: I'd prefer info with an old message. At least during cleanup users won't think that it's stuck somewhere and will get the idea that cleanup is happening.

Comment (tchaton) [reply pytorch_lightning/plugins/training_type/tpu_spawn.py:212]: Let's maybe use something more general as "Finalizing the TPU spawn environment. It shouldn't be long."

Comment (Borda) [reply pytorch_lightning/plugins/training_type/tpu_spawn.py:212]: I would say that finalizing sounds good... without any time estimete

Comment (tchaton) [reply pytorch_lightning/plugins/training_type/tpu_spawn.py:212]: "Finalizing the TPU spawn environment."

Comment (justusschock) [reply pytorch_lightning/plugins/training_type/tpu_spawn.py]: I would vote for debug level. We already have many things that are too verbose IMO.

Comment (awaelchli) [reply pytorch_lightning/plugins/training_type/tpu_spawn.py]: I never said there is a vote haha 🤣 But anyway, this is the current state:

warning: nobody
info: Kaushik, Adrian, Rohit,
debug: Carlos, Justus
unclear: Jirka, Thomas

**Review by ananthsub — APPROVED** [review]

**Pull request merged**